include/Photo.php

```diff
@@ -49,7 +49,7 @@ static function supportedTypes() {
 
 	public function __construct($data, $type=null) {
 		$this->imagick = class_exists('Imagick');
-		$this->types = $this->supportedTypes();
+		$this->types = static::supportedTypes();
```

Comment:
I would say this you need to say `self::` instead of `static::` (`$this->types = self::supportedTypes();`)

Reply:
I copied the idea from here: https://github.com/friendica/friendica/blob/develop/src/App.php#L135

Reply:
@rabuzarus I guess @MrPetovan could explain if this was a good idea. (I hope so)

Reply:
`self` always mean the class where the method is declared, however it isn't always ideal when a child class calls a parent class method. `static` always means the current class which is the intuitive behavior and should be preferred to `self`.

More on PHP's Late Static Binding: http://php.net/manual/en/language.oop5.late-static-bindings.php

Reply:
@MrPetovan thanks for the explanation